### PR TITLE
Add GCP Project Id Text to publish dialogs.

### LINKED ISSUE #791: Publish to appengine flex does not show the name of the target project.
Created a new Google ASP.Net Core Web API project.
Selected my account and my test project (note that first project in the list is a production project).
Ensured that selected project ID also appears in "Enter project ID".
Ensured that appsettings.json contains the correct project ID
Right click project in solution explorer -> Publish to Google Cloud...
Select Appengine Flex
Leave settings unchanged, click Publish

Project compiles successfully

Observe Output window, notice that wrong target project is used.

descriptor:      [C:\Users\xxx\AppData\Local\Temp\c1uamfsa.zjr\app.yaml]
source:          [C:\Users\xxx\AppData\Local\Temp\c1uamfsa.zjr]
target project:  [REDACTED - WRONG project]
target service:  [default]
target version:  [20170904t184514]

Publish fails because I don't have write access to the production project.

### PR DESCRIPTION
**Pull request by ILMTitan**
Fixes #791.

Adds *Google Cloud Project ID* `TextBox` and *Select Project* `Button` to Publish dialogs.

Change project select dialog to change the current project (both for template dialogs and for publish dialogs).

### CODE REVIEW

**Comment by ivannaranjo** [comment]
Would you be interested on parrying down this change to only have the display portions of the project where the deployment is going to happen? That is so we can get the bug fixed and we can have the full discussion on the "current project" model as a separate change?

**Comment by ILMTitan** [comment]
Done. Please take another look.

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/FlexStep/FlexStepContent.xaml [hunk 1]
```diff
@@ -18,6 +18,18 @@
     </UserControl.Resources>
 
     <StackPanel>
+        <DockPanel HorizontalAlignment="Left">
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/FlexStep/FlexStepContent.xaml]: Nit, since these elements are already inside of the `StackPanel` what is the advantage of the nested `DockPanel`?

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/GceStep/GceStepContent.xaml [hunk 2]
```diff
@@ -17,6 +17,18 @@
     </UserControl.Resources>
 
     <StackPanel>
+        <DockPanel HorizontalAlignment="Left">
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/GceStep/GceStepContent.xaml]: Same, question.

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/GkeStep/GkeStepContent.xaml [hunk 3]
```diff
@@ -17,6 +17,18 @@
     </UserControl.Resources>
 
     <StackPanel>
+        <DockPanel HorizontalAlignment="Left">
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/GkeStep/GkeStepContent.xaml]: Same question with the `DockPanel`.

File: GoogleCloudExtension/GoogleCloudExtension/TemplateWizards/Dialogs/ProjectIdDialog/PickProjectIdViewModel.cs [hunk 4]
```diff
@@ -169,6 +169,7 @@ private void OnChangeUser()
 
         private void OnOk()
         {
+            CredentialsStore.Default.UpdateCurrentProject(SelectedProject);
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/TemplateWizards/Dialogs/ProjectIdDialog/PickProjectIdViewModel.cs]: This change doesn't seem relevant to just display the current project.

Comment (ivannaranjo) [reply GoogleCloudExtension/GoogleCloudExtension/TemplateWizards/Dialogs/ProjectIdDialog/PickProjectIdViewModel.cs]: We should not be changing the current project in the selection of the project in the template, we need to find a coherent model for this.

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialog/PublishDialogStepBase.cs [hunk 5]
```diff
@@ -54,6 +55,7 @@ protected set
         }
 
         public abstract FrameworkElement Content { get; }
+        public string GcpProjectId => CredentialsStore.Default.CurrentProjectId;
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialog/PublishDialogStepBase.cs]: Nit, add a comment for the new public property.

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/FlexStep/FlexStepContent.xaml (hunk 1 above)
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/FlexStep/FlexStepContent.xaml]: It was an artifact from when there was also a button to the right of the text box. Now that the button is back, it is needed again.
